webpack/assets/javascripts/foreman_tools.js

```diff
@@ -135,6 +135,11 @@ export function updateTable(element) {
     uri.setSearch('per_page', perPage);
   }
 
+  const searchInput = $('#search-form').find('.autocomplete-input').val();
+  if (searchInput !== undefined && searchInput.trim() === '') {
+    uri.removeSearch('search');
```

Comment:
This looks quite similar to lines 129-134, perhaps time to refactor this function a bit? with this change it will be almost 50 lines long, and is becoming difficult to understand - leading to various cases (such as this) not being handled correctly.

Reply:
indeed, I will take a look at this and try to understand how to clean this mess :grinning: